### PR TITLE
fix(`rpc-types`): use newer `alloy-rpc-types`, `alloy-rpc-trace-types`

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

Fixes the type issues that forced us to use `reth-rpc-types` for the inspector.

## Solution

Use the newer rpc/trace rpc types on alloy and update the reth inspectors which now also use the same types.

### CODE REVIEW

**Review by gakonst — APPROVED** [review]

File: crates/common/src/types.rs [hunk 1]
```diff
@@ -149,11 +149,11 @@ impl ToAlloy for ethers_core::types::Transaction {
     }
 }
 
-impl ToReth for alloy_rpc_types::trace::parity::LocalizedTransactionTrace {
-    type To = alloy_rpc_types::trace::parity::LocalizedTransactionTrace;
+impl ToReth for alloy_rpc_trace_types::parity::LocalizedTransactionTrace {
```
Comment (mattsse) [inline crates/common/src/types.rs:152]: this is now redundant?

File: crates/common/src/types.rs [hunk 2]
```diff
@@ -163,11 +163,11 @@ impl ToReth for alloy_rpc_types::trace::parity::LocalizedTransactionTrace {
     }
 }
 
-impl ToReth for alloy_rpc_types::trace::parity::TransactionTrace {
-    type To = alloy_rpc_types::trace::parity::TransactionTrace;
+impl ToReth for alloy_rpc_trace_types::parity::TransactionTrace {
```
Comment (mattsse) [inline crates/common/src/types.rs:166]: same

**Review by mattsse — CHANGES_REQUESTED** [review]
the ToReth impls are now redundant?

File: crates/common/src/types.rs (hunk 1 above)
Comment (Evalir) [reply crates/common/src/types.rs:152]: true! we can remove this

Comment (Evalir) [reply crates/common/src/types.rs]: removed the `ToReth` trait entirely in https://github.com/foundry-rs/foundry/pull/6705/commits/d8f44c15567cba5be55cdefee0b1c483ee8c798c

File: crates/common/src/types.rs (hunk 2 above)
Comment (Evalir) [reply crates/common/src/types.rs]: removed the `ToReth` trait entirely in https://github.com/foundry-rs/foundry/pull/6705/commits/d8f44c15567cba5be55cdefee0b1c483ee8c798c

**Review by DaniPopes — APPROVED** [review]

**Comment by DaniPopes** [comment]
doctest needs fix on master, clippy still bugged, test fail looks spurious

**Comment by Evalir** [comment]
the integration test failure is not spurious but rather seems to be an edge case when fetching uncle blocks on anvil—this can be fixed separately in another PR, don't think it should block merging this one
